route_guide/route_guide_server.cc

```diff
@@ -55,133 +64,235 @@ float GetDistance(const Point& start, const Point& end) {
+// which uses callbacks. stout-grpc is intented to be a higher-level interface
+// while still being asynchronous that uses stout-eventuals
+// (https://github.com/3rdparty/stout-eventuals) to make composing async code
+// easier.
+// This class contains 4 methods: ServeGetFeature(unary rpc), ServeListFeatures
+// (server streaming), ServeRecordRoute(client streaming), ServeRouteChat
+// (bidirectional streaming).
 class RouteGuideImpl final {
  public:
+  // Populate feature_list_ vector with addresses from database
+  // (route_guide_db.json)
   explicit RouteGuideImpl(const std::string& db) {
     routeguide::ParseDb(db, &feature_list_);
   }
 
-  auto Serve(Server* server) {
-    auto status = server->Serve<RouteGuide, Point, Feature>(
-        "GetFeature",
-        [this](auto* call, auto&& point) {
-          Feature feature;
-          feature.set_name(GetFeatureName(*point, feature_list_));
-          feature.mutable_location()->CopyFrom(*point);
-          call->WriteAndFinish(feature, Status::OK);
-        },
-        [](auto*, bool) {});
-
-    if (!status.ok()) {
-      return status;
-    }
+  // The goal of this function is to serve unary rpc call. Client sends to the
+  // server the request which contains Point's coordinates (longitude,
+  // latitude). The server checks received Point's coordinates, if database has
+  // the specific address which corresponds to Point's coordinates - the server
+  // will response with it address.
+  // Accept<...>() method (stout/grpc/server.h) gets the name of rpc call and
+  // the host name and returns the Composable structure composed with Map()
+  // using overloaded operator "|". Then we compose with Map() in order to
+  // have deal with multiple connections and finally compose with Loop() to
+  // convert stream into the eventual.
+  auto ServeGetFeature(Server* server) {
+    return server->Accept<
+               RouteGuide, // Service name
+               Point, // Request from client
+               Feature>( // Response from the server
+               "GetFeature")
+        | Map(Then([this](auto&& context) {
+             // Server::Handler() function returns Composable structure, so it
+             // can be composed with other tasks. .body() callback will be
+             // invoked every time the client will send the request to the
+             // server.
+             return Server::Handler(std::move(context))
+                 .body([this](auto& call, auto&& point) {
+                   // Check if the request is valid
+                   if (point) {
+                     std::cout << "Server is preparing the response..."
+                               << std::endl;
+                     std::cout << "The request is: ["
+                               << point.get()->latitude() << ", "
+                               << point.get()->longitude() << "]"
+                               << std::endl;
```

Comment:
Does the original gRPC example print this much to `std::cout`? Let's try to match (only) the `std::cout` statements that the original has, to (1) show how we're the same program just written differently, and (2) avoid spamming our users with more info than they want.